src/mono/mono/eventpipe/ds-protocol.c

```diff
@@ -347,7 +348,7 @@ ds_ipc_message_try_parse_value (
 
 	memcpy (value, *buffer, value_len);
 	*buffer = *buffer + value_len;
-	*buffer_len = *buffer_len - value_len;
+	*buffer_len = *buffer_len - (uint32_t)value_len;
```

Comment:
Why do you cast here but not in the `EP_ASSERT` just above?

Also, just noticed - the assert is wrong - should be `EP_ASSERT ((*buffer_len - value_len) >= 0);`
Also both types are unsigned, so I think the result is unsigned, which means the comparison is always true.


Also how did we end up with this mix of `size_t` and `uint32_t` ?

Reply:
Will drop the assert or do assert that checks for underflow. I ended up with size_t since callers use sizeof(x) as parameter, so instead of casting at call sites I handled in this function. The uint32 is used since C++ library is using either uint16 or uint32 in its template methods, so I standardized on the bigger, but didn't move all the way to size_t.

Reply:
But number of callers is small, so I can just pick uint32 instead of size_t and simplify it a little. Will do that change.

Reply:
Fixed assert and changed type to uint32_t and cast at call sites .